adal/tests/Test.ADAL.NET.UIAutomation/Infrastructure/KeyVaultSecretsProvider.cs

```diff
@@ -0,0 +1,98 @@
+﻿
+using Microsoft.Azure.KeyVault;
+using Microsoft.Azure.KeyVault.Models;
+using Microsoft.Extensions.Configuration;
+using Microsoft.IdentityModel.Clients.ActiveDirectory;
+using System;
+using System.Threading.Tasks;
+
+namespace Test.ADAL.NET.UIAutomation
+{
+    public class KeyVaultSecretsProvider
+    {
+        /// <summary>
+        /// Token cache used by the test infrastructure when authenticating against KeyVault
+        /// </summary>
+        /// <remarks>We aren't using the default cache to make sure the tokens used by this
+        /// test infrastructure can't end up in the cache being used by the tests (the UI-less
+        /// Desktop test app runs in the same AppDomain as the infrastructure and uses the
+        /// default cache).</remarks>
+        private readonly static TokenCache keyVaultTokenCache = new TokenCache();
+
+        private KeyVaultClient _keyVaultClient;
+
+        private ClientAssertionCertificate _assertionCert;
+
+        private KeyVaultConfiguration _config;
+
+        private readonly string keyVaultClientID = "ebe49c8f-61de-4357-9194-7a786f6402b4";
+
+        private readonly string keyVaultThumbPrint = "440A5BE6C4BE2FF02A0ADBED1AAA43D6CF12E269";
```

Comment:
You might want to run our internal compliance build before commiting, I think it might fail as it will consider there to be internal.